packages/open-next/src/utils/regex.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Constructs a regular expression for a path that supports separators for multiple platforms
+ *  - Uses posix separators (`/`) as the input that should be made cross-platform.
+ *  - Special characters are escaped by default but can be controlled through opts.escape.
+ *  - Posix separators are always escaped.
+ *
+ * @example
+ * ```ts
+ * getCrossPlatformPathRegex("./middleware.mjs")
+ * getCrossPlatformPathRegex(String.raw`\./middleware\.(mjs|cjs)`, { escape: false })
+ * ```
+ */
+export function getCrossPlatformPathRegex(
+  regex: string,
+  opts: { escape: boolean } = { escape: true },
+) {
+  const newExpr = (
+    opts.escape ? regex.replace(/([[\]().*+?^$|{}\\])/g, "\\$1") : regex
+  ).replaceAll("/", String.raw`(?:\/|\\)`);
+
+  return new RegExp(newExpr, "g");
```

Comment:
I don't think we really need "g" here?
At least it doesn't harm.
Do we want to be able to set that via `{escape = true, flags = "g"}`